tiatoolbox/models/engine/patch_predictor.py

```diff
@@ -14,46 +14,51 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 #
-# The Original Code is Copyright (C) 2021, TIA Centre, University of Warwick
+# The Original Code is Copyright (C) 2021, TIALab, University of Warwick
 # All rights reserved.
 # ***** END GPL LICENSE BLOCK *****
 
 """This module implements patch-level prediction."""
 
+import copy
 import os
 import pathlib
 import warnings
 from collections import OrderedDict
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Union
 
 import numpy as np
 import torch
 import tqdm
 
-from tiatoolbox.models.abc import IOConfigABC
 from tiatoolbox.models.architecture import get_pretrained_model
 from tiatoolbox.models.dataset.classification import PatchDataset, WSIPatchDataset
+from tiatoolbox.models.engine.semantic_segmentor import IOSegmentorConfig
 from tiatoolbox.utils import misc
 from tiatoolbox.utils.misc import save_as_json
 from tiatoolbox.wsicore.wsireader import VirtualWSIReader, get_wsireader
 
 
-class IOPatchPredictorConfig(IOConfigABC):
+class IOPatchPredictorConfig(IOSegmentorConfig):
```

Comment:
It would be best to actually bring IOSegmentorConfig functionality to ABC. However, I will do that in another PR as it will require testing and refactoring other stuffs. This one focuses on fixing the bug only atm .